components/Search/serialize.js

```diff
@@ -0,0 +1,45 @@
+import { SUPPORTED_FILTER, SUPPORTED_SORT } from './constants'
+
+const isSupportedFilter = filter => {
+  return (
+    !!SUPPORTED_FILTER[filter.key] &&
+    SUPPORTED_FILTER[filter.key].indexOf(filter.value) !== -1
+  )
+}
+
+export const deserializeFilters = filtersString => {
+  return decodeURIComponent(filtersString)
+    .split('|')
+    .filter(filter => filter.split(':').length === 2)
+    .map(filter => ({ key: filter.split(':')[0], value: filter.split(':')[1] }))
+    .filter(filter => isSupportedFilter(filter))
+}
```

Comment:
I think this would improve readability:
```
  return decodeURIComponent(filtersString)
    .split('|')
    .map(f => f.split(':'))
    .filter(f => f.length === 2)
    .map(([key, value]) => ({ key, value }))
    .filter(f => isSupportedFilter(f))
```